pkg/cephfs/nodeserver.go

```diff
@@ -308,6 +309,89 @@ func (ns *NodeServer) NodeUnstageVolume(ctx context.Context, req *csi.NodeUnstag
 	return &csi.NodeUnstageVolumeResponse{}, nil
 }
 
+func (ns *NodeServer) NodeGetVolumeStats(ctx context.Context, req *csi.NodeGetVolumeStatsRequest) (*csi.NodeGetVolumeStatsResponse, error) {
+
+	var err error
+	targetPath := req.GetVolumePath()
+	if targetPath == "" {
+		err = fmt.Errorf("targetpath %v is empty", targetPath)
+		return nil, status.Error(codes.InvalidArgument, err.Error())
+	}
+	volID := req.GetVolumeId()
+	/* TODO: Map the volumeID to the targetpath.
+
+	we need secret to connect to the ceph cluster to get the volumeID from volume
+	Name, however `secret` field/option is not available  in NodeGetVolumeStats spec,
+	Below issue covers this request and once its available, we can do the validation
+	as per the spec.
+	https://github.com/container-storage-interface/spec/issues/371
+
+	*/
+
+	isMnt, err := isMountPoint(targetPath)
+
+	if err != nil {
+		klog.Errorf("stat failed on path: %v, error: %v", targetPath, err)
+		return nil, status.Error(codes.InvalidArgument, err.Error())
+	}
+
+	if isMnt {
+		klog.Infof("cephfs: volume %s is already mounted to %s", volID, targetPath)
+		cephfsProvider := volume.NewMetricsStatFS(targetPath)
```

Comment:
Error check to ensure `cephfsProvider` is non-nil?

Reply:
this is not required. never going to happen

Reply:
yep, this is not gonna happen.

Reply:
Is this because the signature does not have a return like (xxx, err) that we can assume this will not fail? Attempting to understand why this is not going to happen? Not a blocker for merge, but looking for information.